function-sidecar/pkg/dispatcher/grpc/grpc.go

```diff
@@ -147,3 +258,43 @@ func toDispatcher(grpc *function.Message) message.Message {
 	}
 	return message.NewMessage(grpc.Payload, dHeaders)
 }
+
+func unmarshallFactory() WindowingStrategyFactory {
+	var w v1alpha1.Windowing
+	json.Unmarshal([]byte(os.Getenv("WINDOWING_STRATEGY")), &w)
+
+	strategies := 0
+	v := reflect.ValueOf(w)
+	for i := 0; i < v.NumField(); i++ {
+		f := v.Field(i)
+		if f.Interface() != reflect.Zero(f.Type()).Interface() {
+			strategies++
+		}
+	}
+
+	if strategies > 1 {
+		panic("windowing strategies are mutually exclusive")
+	}
+
+	if w.Size != int32(0) {
+		log.Printf("Will use windowing strategy of %v messages\n", w.Size)
+		return sizeFactoryFactory(int(w.Size))
+	} else if w.Time != "" {
+		log.Printf("Will use time based windowing strategy of %v\n", w.Time)
+		d, err := time.ParseDuration(w.Time)
+		if err != nil {
+			panic(err) // Assume it's validated at fn registration time
+		}
+		return ptimeFactoryFactory(d)
+	} else if w.Session != "" {
+		log.Printf("Will use session based windowing strategy of %v\n", w.Session)
+		d, err := time.ParseDuration(w.Session)
+		if err != nil {
+			panic(err) // Assume it's validated at fn registration time
+		}
+		return ptimeSessionFactoryFactory(d)
+	} else {
+		log.Printf("Will use no windowing strategy (unbounded stream)\n")
```

Comment:
I'd prefer this to say "Will use unbounded windowing strategy ..." since we are using a strategy even though it's unbounded.

Reply:
Will change